runtime/caml/roots.h

```diff
@@ -29,9 +29,11 @@ intnat caml_darken_all_roots_slice (intnat);
 void caml_do_roots (scanning_action, int);
 extern uintnat caml_incremental_roots_count;
 #ifndef NATIVE_CODE
+#define caml_do_local_roots caml_do_local_roots_byt
 CAMLextern void caml_do_local_roots (scanning_action, value *, value *,
                                      struct caml__roots_block *);
 #else
+#define caml_do_local_roots caml_do_local_roots_nat
 CAMLextern void caml_do_local_roots(scanning_action f, char * c_bottom_of_stack,
                                     uintnat last_retaddr, value * v_gc_regs,
                                     struct caml__roots_block * gc_local_roots);
```

Comment:
If this ends up going in some form, I think it might be clearer to rename the functions in `roots_byt.c` and `roots_nat.c` to the `_byt` and `_nat` versions and put the `#define` after the `CAMLextern` using the actual name!

Reply:
Good idea, done